apollo-gradle-plugin/src/main/kotlin/com/apollographql/apollo/gradle/internal/ApolloDownloadSchemaTask.kt

```diff
@@ -1,36 +1,32 @@
 package com.apollographql.apollo.gradle.internal
 
-import com.squareup.moshi.JsonWriter
-import okhttp3.HttpUrl.Companion.toHttpUrl
-import okhttp3.MediaType.Companion.toMediaTypeOrNull
-import okhttp3.OkHttpClient
-import okhttp3.Request
-import okhttp3.RequestBody.Companion.toRequestBody
-import okio.buffer
-import okio.sink
 import org.gradle.api.DefaultTask
-import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.ListProperty
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.TaskAction
-import java.io.ByteArrayOutputStream
-import java.util.concurrent.TimeUnit
+import org.gradle.api.tasks.options.Option
+import java.io.File
 
 abstract class ApolloDownloadSchemaTask : DefaultTask() {
   @get:Input
-  abstract val endpointUrl: Property<String>
+  @get:Option(option = "endpoint", description = "url of the GraphQL endpoint")
+  abstract val endpoint: Property<String>
 
   @get:Input
-  abstract val schemaFilePath: Property<String>
+  @get:Optional
+  @get:Option(option = "schema", description = "path where the schema will be downloaded, relative to the current working directory")
+  abstract val schema: Property<String>
 
   @get:Optional
   @get:Input
-  abstract val headers: MapProperty<String, String>
+  @set:Option(option = "header", description = "headers in the form 'Name: Value'")
+  var header = emptyList<String>() // cannot be lazy for @Option to work
 
-  @get:Optional
   @get:Input
-  abstract val queryParameters: MapProperty<String, String>
+  @get:Optional
+  abstract val schemaRelativeToProject: Property<String>
```

Comment:
@martinbonnin is there a particular reason why this property doesn't have the `@Option` annotation?

Reply:
Not really. The reasoning was that when run from the command line, most tools assume parameters to be relative the current working directory. But adding `@Option` here works as well. Feel free to make a pull request. Or I can do it if you think that'll be useful.

Reply:
I just wanted to make sure that I wasn't missing something.
I can do it, thanks!

Reply:
I was looking into this this morning so I made a PR there: https://github.com/apollographql/apollo-android/pull/2378